jena-text/src/main/java/org/apache/jena/query/text/TextQueryPF.java

```diff
@@ -319,13 +311,13 @@ private StrMatch objectToStruct(PropFuncArg argObject, boolean executionTime) {
 
             String lang = o.getLiteralLanguage();
             RDFDatatype dt = o.getLiteralDatatype() ;
-            if (noLang(lang)) {
+            if (lang.isEmpty()) {
                 if (dt != null && dt != XSDDatatype.XSDstring) {
                     log.warn("Object to text query is not a string") ;
                     return null ;
                 }
             }
-            lang = fixLang(lang);
+            lang = lang.isEmpty() ? null : lang;
```

Comment:
I like [`emptyToNull`](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Strings.html#emptyToNull(java.lang.String)) for this, but this isn't a problem or anything.

Reply:
Agreed and changed. Thanks